nautobot/extras/tests/example_jobs/test_atomic_transaction.py

```diff
@@ -0,0 +1,44 @@
+from django.db import transaction
+
+from nautobot.core.celery import register_jobs
+from nautobot.extras.jobs import Job, BooleanVar
+from nautobot.extras.models import Status
+
+
+class TestAtomicDecorator(Job):
+    """
+    Job that uses @transaction.atomic decorator to roll back changes.
+    """
+
+    fail = BooleanVar()
+
+    @transaction.atomic
+    def run(self, fail=False):
+        try:
+            Status.objects.create(name="Test database atomic rollback 1")
+            if fail:
+                raise Exception("simulated failure")
+        except Exception:
+            self.log_failure("Job failed, all database changes have been rolled back.")
```

Comment:
Does this work? I'd have thought that since we're catching the exception, the rollback won't actually happen?

Reply:
`log_failure` raises another exception which causes the transaction to roll back. This is tested in `nautobot/extras/tests/test_jobs:JobTest`